File: quizzes9/12-problem.ts
export {};

const isPrimeNumber = (num: number): boolean => {
    if (!Number.isInteger(num) || num < 2) return false;
    if (num === 2) return true;
    if (num % 2 === 0) return false;

    const limit: number = Math.sqrt(num);

    for (let i = 3; i <= limit; i += 2) {
        if (num % i === 0) return false;
    }

    return true;
};

console.log(isPrimeNumber(9)); // "false"
console.log(isPrimeNumber(25)); // "false"
console.log(isPrimeNumber(65521)); // "true" (Largest prime below 2^16)
console.log(isPrimeNumber(19)); // "true"
console.log(isPrimeNumber(1)); // "false"
console.log(isPrimeNumber(17.5)); // "false"

File: bins/binsV2/bin/bins6/12-problem.rb
# frozen_string_literal: true

def is_prime_number(num)
  return false unless num.is_a?(Integer) && num >= 2
  return true if num === 2
  return false if num % 2 === 0

  limit = Math.sqrt(num)
  i = 3

  while i <= limit
    return false if num % i === 0
    i += 2
  end

  true
end

puts is_prime_number(9)      # false
puts is_prime_number(25)     # false
puts is_prime_number(65521)  # true  (Largest prime below 2^16)
puts is_prime_number(19)     # true
puts is_prime_number(1)      # false
puts is_prime_number(17.5)   # false
